pkg/adapter/mtping/runner.go

```diff
@@ -140,16 +132,43 @@ func (a *cronJobsRunner) cronTick(ctx context.Context, event cloudevents.Event)
 	}
 }
 
-type message struct {
-	Body string `json:"body"`
-}
+func makeEvent(source *v1beta2.PingSource) cloudevents.Event {
+	event := cloudevents.NewEvent()
+	event.SetType(v1beta2.PingSourceEventType)
+	event.SetSource(v1beta2.PingSourceSource(source.Namespace, source.Name))
+	if source.Spec.CloudEventOverrides != nil && source.Spec.CloudEventOverrides.Extensions != nil {
+		for key, override := range source.Spec.CloudEventOverrides.Extensions {
+			event.SetExtension(key, override)
+		}
+	}
 
-func makeMessage(body string) interface{} {
-	// try to marshal the body into an interface.
-	var objmap map[string]*json.RawMessage
-	if err := json.Unmarshal([]byte(body), &objmap); err != nil {
-		// Default to a wrapped message.
-		return message{Body: body}
+	// Set event data, at most one of data and dataBase64 exists.
+	// 1. If dataBase64 exists, then it's binary data, set event.DataEncoded to []byte(dataBase64)
+	// 2. If data exists, then it's not binary data
+	//  a. If contentType is not `application/json`, set event.DataEncoded to []byte(data)
+	//  b. If contentType is `application/json`, unmarshal it into an interface, event.DataEncoded will be json.Marshal(interface),
+	//    this is to be compatible with the existing v1beta1 PingSource -> CloudEvent conversion logic, to make sure
+	//    that `data` is populated in the cloudevent json format instead of `data_base64`, and not breaking subscribers
+	//    that does not leverage cloudevents sdk.
+	var data interface{}
+	if source.Spec.DataBase64 != "" {
+		data = []byte(source.Spec.DataBase64)
+	} else if source.Spec.Data != "" {
+		switch source.Spec.ContentType {
+		case cloudevents.ApplicationJSON:
+			// unmarshal the body into an interface, JSON validation is done in pingsource_validation
+			// ignoring the error returned by json.Unmarshal here.
+			var objmap map[string]*json.RawMessage
+			_ = json.Unmarshal([]byte(source.Spec.Data), &objmap)
```

Comment:
might be worth logging the error?

Reply:
thanks for bring this up! imo only logging is still "swallowing" the error, and user can get broken CloudEvent sent to downstream, in PingSource there was no failure mode in `AddSchedule`, @lionelvillard do you have any thoughts?